### ISSUE TITLE
[BUG] Recipe hyperparameter tuning bug

### ISSUE THREAD

**Issue by dan-licht** [mlflow/mlflow#8859]
### Issues Policy acknowledgement

- [X] I have read and agree to submit bug reports in accordance with the [issues policy](https://www.github.com/mlflow/mlflow/blob/master/ISSUE_POLICY.md)

### Willingness to contribute

Yes. I would be willing to contribute a fix for this bug with guidance from the MLflow community.

### MLflow version

- Client: 2.4.1

### System information

- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Databricks Runtime 11.3.x-gpu-ml-scala2.12
- **Python version**: 3.9.5

### Describe the problem

I am using the hyperopt hyperparameter tuning functionality for a classification model. In the logs I see:
```
WARNING mlflow.recipes.steps.train: Failed to build tuning results table due to unexpected failure: 'NoneType' object has no attribute 'keys'
```
Upon examining the source code, it looks like the warning stems from [L599 here](https://github.com/mlflow/mlflow/blob/master/mlflow/recipes/steps/train.py#L599). `best_estimator_params` is initialized as `None` [on line 368](https://github.com/mlflow/mlflow/blob/master/mlflow/recipes/steps/train.py#L368), but it is not updated by any of the ensuing code until the `.keys()` call is made on line 599, causing the error.

Note, the training still runs, I believe this just impacts the training html card, which does not show the hyperparameter tuning trials tab.

### Code to reproduce issue

```python
from mlflow.recipes import Recipe
r = Recipe(profile="databricks")
r.run("train")
```

### Stack trace

```
WARNING mlflow.recipes.steps.train: Failed to build tuning results table due to unexpected failure: 'NoneType' object has no attribute 'keys'
```

### Other info / logs

```
REPLACE_ME
```

### What component(s) does this bug affect?

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [X] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

### What interface(s) does this bug affect?

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

### What language(s) does this bug affect?

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

### What integration(s) does this bug affect?

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [X] `integrations/databricks`: Databricks integrations

**Comment by WeichenXu123** [comment]
> Upon examining the source code, it looks like the warning stems from [L599 here](https://github.com/mlflow/mlflow/blob/master/mlflow/recipes/steps/train.py#L599). best_estimator_params is initialized as None [on line 368](https://github.com/mlflow/mlflow/blob/master/mlflow/recipes/steps/train.py#L368), but it is not updated by any of the ensuing code until the .keys() call is made on line 599, causing the error.


Yes. I agree with your root cause analysis. Feel free to file PR !
